### PR TITLE
Don't save recordings for failed tests

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Also add back SaveDebugRecordingsOnFailure that @heaths added prior to Test Proxy migration, now that there is a way to control whether or not to save in Test Proxy.

### CODE REVIEW

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 1]
```diff
@@ -97,6 +97,28 @@ public abstract class RecordedTestBase : ClientTestBase
         /// </summary>
         public List<(string Header, string QueryParameter)> SanitizedQueryParametersInHeaders { get; } = new();
 
+        /// <summary>
+        /// Flag you can (temporarily) enable to save failed test recordings
+        /// and debug/re-run at the point of failure without re-running
+        /// potentially lengthy live tests.  This should never be checked in
+        /// and will throw an exception from CI builds to help make that easier
+        /// to spot.
+        /// The list of JSON path sanitizers to use when sanitizing a JSON request or response body.
```
Comment (christothes) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:106]: Is this last comment relevant to this setting? If so, could it be clarified?

File: sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs [hunk 2]
```diff
@@ -249,6 +249,10 @@ internal HttpMessage CreateStopRecordRequest(string xRecordingId, IDictionary<st
             uri.AppendPath("/record/stop", false);
             request.Uri = uri;
             request.Headers.Add("x-recording-id", xRecordingId);
+            if (xRecordingSkip != null)
+            {
+                request.Headers.Add("x-recording-skip", xRecordingSkip);
```
Comment (christothes) [inline sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs:254]: What values would be valid for `xRecordingSkip`? If it's just a `bool`, should we make it a `bool` and `ToString` it here?

File: sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs [hunk 3]
```diff
@@ -265,9 +269,10 @@ internal HttpMessage CreateStopRecordRequest(string xRecordingId, IDictionary<st
         /// <summary> Stop recording for a test. </summary>
         /// <param name="xRecordingId"> The recording ID. </param>
         /// <param name="variables"> Variables for the recording. </param>
+        /// <param name="xRecordingSkip"> Set to request-response to skip recording this session. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="xRecordingId"/> or <paramref name="variables"/> is null. </exception>
-        public async Task<Response> StopRecordAsync(string xRecordingId, IDictionary<string, string> variables, CancellationToken cancellationToken = default)
+        public async Task<Response> StopRecordAsync(string xRecordingId, IDictionary<string, string> variables, string xRecordingSkip = null, CancellationToken cancellationToken = default)
```
Comment (christothes) [inline sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs:275]: Should `xRecordingSkip` be a bool?

Comment (christothes) [inline sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs:275]: Is this new argument plumbed through to `TestRecording` somewhere?

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:106]: Woops will remove.

File: sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs:254]: It could be a bool, but the Test Proxy is expecting "request-response" as the value, and this is generated code.

File: sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs:275]: Test Proxy expects "request-response" as the value here.

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/Generated/TestProxyRestClient.cs:275]: It is passed to this method from DisposeAsync in TestRecording.

**Review by christothes — APPROVED** [review]
